pkg/apis/v1beta1/nodepool_status.go

```diff
@@ -23,4 +23,35 @@ type NodePoolStatus struct {
 	// Resources is the list of resources that have been provisioned.
 	// +optional
 	Resources v1.ResourceList `json:"resources,omitempty"`
+	// Conditions represent the latest available observations of a NodePool's current state.
+	// +optional
+	Conditions []NodePoolCondition `json:"conditions,omitempty"`
+}
+
+// NodePoolConditionType represents a NodePool condition value.
+type NodePoolCondition struct {
```

Comment:
There's already a generalized condition type that we use from knative. See the conditions field in the NodeClaim